CP-34198: Start membership naming for pool hosts

psafont opened the pull request:
Pool hosts are now called 'members', with one of them being the master
host.

There's no consensus on what to do with the term master, but this naming
scheme allows us to stop using the term slave.

Citrix's writing guides is not very self-explanatory,it would be interesting if people would propose alternatives to "master" in any case.
 https://brand.citrix.com/application-guides/writing/unbiased-language/#section_10

lindig commented:
I think it's important to state whether semantics change or not. In my understanding a slave is a member but not a master. Whereas any host (master or not) is a member of a pool.

lippirk commented:
I would also be happy with master -> primary, slave -> secondary like the branding says, but slave -> member is probably fine

edwintorok commented on ocaml/xapi/xapi_cluster.ml:
```diff
@@ -151,7 +153,7 @@ let pool_destroy_common ~__context ~self ~force =
         List.filter (( <> ) master_ch) all_hosts
   in
   foreach_cluster_host ~__context ~self ~log:force
-    ~fn:Client.Client.Cluster_host.destroy slave_cluster_hosts
```
There is another instance of `slave` here

psafont replied:
Yes, there are a lot of mentions to slaves around the whole codebase. I concentrated the efforts of this PR around xapi_pool_helpers and its users. I can extend it a bit more if we find the terminology good enough to use.

psafont commented:
Converted the changes to use primary / secondary as it's being picked up by more people. I've also replaced a lot of usaged. There are about 55 more files with mentions to slave, there are quite a few that cannot be removed because they affect the API or because they don't refer to host but to other concepts that don't involve host (like bridges), which we don't have control of.

This is a starting pass and not all usages have been changed. The networking module specially is difficult to change because it mixes different types of "slaves" and it's not clear if the name or comment is referring to hosts or not.

psafont commented:
Drafting this as a decisions needs to be reached among the whole hypervisor team about which terms to use

psafont commented:
Closing this, it's obvious we don't have capacity for this currently.
